schemas/Composer.json

```diff
@@ -20,8 +20,20 @@
     "startsThread": {
       "description": "Whether this message begins a new thread or not.",
       "type": "boolean"
+    },
+    "isRichTextEditor": {
+      "description": "Whether this message was composed in the rich text editor (as opposed to the predating markdown-based editor).",
+      "type": "boolean"
+    },
+    "isRichTextEditorFormattingEnabled": {
+      "description": "Whether the rich text editor is in rich or plain text mode. This value is not applicable (always false) if isRichTextEditor is false.",
+      "type": "boolean"
+    },
+    "isMarkdownEnabled": {
+      "description": "Whether markdown is supported in the editor. This value is not applicable (always false) if isRichTextEditorFormattingEnabled is true.",
+      "type": "boolean"
     }
```

Comment:
This is a good point. However it's a bit complicated because the markdown setting affects the legacy composer and the rich text editor (when it's in plain text mode).

If we replicate the modes and markdown setting as they are in the app UI, we'd end up with:

- `composer: LEGACY | RTE_FORMATTING | RTE_PLAIN`
- `markdown: true | false`

The caveat being that `markdown` does not apply when `composer` is `RTE_FORMATTING`, but I think the problem is not in the analytics model but rather that the app lacks that feature.

Reply:
Sounds good, will update.